### PR TITLE
Replica exchange + metadynamics

### LINKED ISSUE #1: REMD broken with GLE thermostats
Allegedly REMD gets messed up when using a GLE thermostat. I think this actually makes sense, because the s are not rescaled after the temperature swap and so they sample the wrong ensemble. Should be an easy fix I'll probably incorporate it in a REMD+METAD branch I'm working on in the cosmo-epfl fork

### PR DESCRIPTION
**Pull request by ceriottm**
Using PLUMED. Relies on bias weights to activate different biases on the various replicas. Requires multiple FFPlumed to be defined, which can be a bit bothersome but I don't think this will be used often enough to warrant creating a template structure. That said, it wouldn't be too hard to create a ff_template mimicking system_template.
Also, fixes #1

### CODE REVIEW

File: examples/regtest/ph2/remd-gle/our_ref.pdb [hunk 1]
```diff
@@ -0,0 +1,111 @@
+TITLE   cell{angstrom} Traj: positions{angstrom} Step:           0  Bead:       1
```
Comment (mahrossi) [inline examples/regtest/ph2/remd-gle/our_ref.pdb]: Do we really want to keep PDB files in the examples?

File: ipi/engine/forcefields.py [hunk 2]
```diff
@@ -174,12 +168,16 @@ def queue(self, atoms, cell, reqid=-1):
     def poll(self):
         """Polls the forcefield object to check if it has finished."""
 
-        for r in self.requests:
-            if r["status"] == "Queued":
-                r["t_dispatched"] = time.time()
-                r["result"] = [0.0, np.zeros(len(r["pos"]), float), np.zeros((3, 3), float), ""]
-                r["status"] = "Done"
-                r["t_finished"] = time.time()
+        self._threadlock.acquire()
```
Comment (mahrossi) [inline ipi/engine/forcefields.py]: ok, this is also in the other pull request.

File: examples/regtest/ph2/remd-gle/our_ref.pdb (hunk 1 above)
Comment (ceriottm) [reply examples/regtest/ph2/remd-gle/our_ref.pdb]: No strong feelings but I think most p-h2 examples use PDB. I'd say unrelated to this PR

File: ipi/engine/forcefields.py (hunk 2 above)
Comment (ceriottm) [reply ipi/engine/forcefields.py]: yep now it's merged

**Pull request merged**